chapter07/chapter07.adoc

```diff
@@ -0,0 +1,276 @@
+
+==== Example
+
+Consider a metric that tracks the number of user login attempts in a web application. The metadata for this metric might look like this:
+
+- Name: `login_attempts`
+- Description: `"Counts the number of user login attempts, including successes and failures."`
+- Type: `Counter`
+- Unit: `attempts`
+- Tags: `{ action="login", outcome="success|failure" }`
+
+This metadata tells us that `login_attempts` is a counter metric, meaning it will increment over time, tracking login attempts. The description provides context on what is being counted, and the tags allow for differentiation between successful and failed attempts.
+
+==== Purpose and Importance of Metadata
+
+- *Identification*: Metadata helps uniquely identify a metric within the application or across different systems. The name and tags associated with a metric can precisely pinpoint its source and purpose.
+- *Documentation*: Through descriptive texts and labels, metadata serves as documentation for metrics, explaining their meaning, usage, and any relevant context that users might need to interpret the data correctly.
+- *Standardization*: By defining expected metadata for metrics, MicroProfile Metrics promotes standardization across applications and tools, facilitating easier integration with monitoring and analysis tools.
+- *Categorization and Filtering*: Metadata allows metrics to be categorized and filtered based on tags or other attributes. This is particularly useful in complex systems with numerous metrics, enabling users to focus on the most relevant data points.
+- *Configuration*: Some metadata fields can influence how a metric is collected, stored, or displayed by monitoring systems, offering a degree of control over the metrics pipeline.
+
+=== Categories of Metrics
+
+In MicroProfile Metrics, metrics are organized into three distinct scopes: Base, Vendor, and Application. This categorization is designed to clearly separate metrics by their origin and relevance, making it easier for developers and operators to monitor and manage the performance of their microservices. Each scope serves a specific purpose and contains a different set of metrics:
+
+- *Base Metrics* are common to all applications, such as the number of CPUs or the amount of free memory. These metrics provide essential information about the underlying Java Virtual Machine (JVM) and the core libraries that are common across all MicroProfile applications. Base metrics typically include JVM-specific metrics such as memory usage, CPU load, thread counts, and garbage collection statistics. The intention behind base metrics is to offer a consistent set of low-level metrics that are universally applicable and useful for monitoring the health and performance of the JVM itself, which is the foundation upon which all MicroProfile applications run. 
+Base metrics are exposed under the path `/metrics?scope=base`.
```

Comment:
In 5.x, base metrics have become optional. Not sure if that's worth mentioning for the purpose of the tutorial.

Reply:
Added a note stating that Base metrics has become optional in 5.x